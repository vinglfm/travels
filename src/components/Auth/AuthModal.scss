$focusColor: #88D5E9;
$submitColor: #3b5998;
$inputColor: #BEBEBE;
$errorColor: red;

@mixin auth-input($color) {
  box-sizing: border-box;
  border: 1px solid $inputColor;
  padding: .3rem;
  width: 97%;
  font-size: 1.2rem;
  margin: 0;
  outline: none;

    &:focus {
      -moz-box-shadow: 0 0 8px $color;
      -webkit-box-shadow: 0 0 8px;
      box-shadow: 0 0 8px $color;
      border: 1px solid $color;
    }
}

:local .dialog__title {
  text-align: center;
  font-size: 2rem;
}

:local .fb__button {
    font-family: Helvetica, sans-serif;
    font-weight: 500;
    font-size: 2rem;
    background-color: #4c69ba;
    color: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    cursor: pointer;
    display: inline-block;
    text-decoration: none;
    text-transform: uppercase;
    background-color: #4c69ba;
    padding: .6rem;
}

:local .lines {
    margin: 1rem;
    line-height: 0.5;
    text-align: center;
    font-family: Helvetica, sans-serif;
    font-weight: 500;
    font-size: 2rem;

    span {
        display: inline-block;
        position: relative;  

        &:before,
        &:after {
            content: "";
            position: absolute;
            height: 0.1rem;
            border-bottom: 1px solid black;
            border-top: 1px solid black;
            top: 50%;
            width: 3rem;
        }

        &:before {
            right: 150%;
            margin-right: 2px;
        }

        &:after {
            left: 150%;
            margin-left: 2px;
        }
    }
  }

:local .log__in {
    font-style: italic;
    text-align: center;
    a {
        font-weight: bold;
        font-size: 2rem;
        cursor: pointer;
    }
}

:local .sign__up {
  font-style: italic;
  font-size: 1rem;

  a {
    font-weight: bold;
    font-size: 1.5rem;
    cursor: pointer;
  }
}

:local .sign__in {
  text-align: center;
}

:local .modal {
    font: 1rem "Lucida Sans Unicode", "Lucida Grande", sans-serif;

    &__back {
      float: left;
      cursor: pointer;
    }
    
    &__header {
      text-align: center;
      font-size: 1.5rem;
      font-weight: bold;
    }
  
    &__divider {
      border-width: .5px;
      border-color: $inputColor;
    }
  
    &__form {
      margin: .3rem auto;
      display: inline-block;
      width: 100%;

      &__input {
        @include auth-input($focusColor);
      }

      &__input__error {
        @include auth-input($errorColor);
      }
  
      &__label {
        font-size: 1rem;
        margin:0 0 3px 0;
        padding: 0;
        display: block;
        font-weight: bold;
  
        &__required {
          color: $errorColor;
        }
      }

      &__error {
        position: relative;
        overflow: hidden;
        max-width: 95%;
        min-height: 1rem;

        &__message {
          position: absolute;
          font-size: .6rem;
          color: $errorColor;
          white-space: nowrap;

          &__hidden {
            display: none;
          }
        }
      }
      
  
      &__input__small {
        @extend .modal__form__input;
        width: 48%;
      }
  
      &__group {
        text-align: center;
        padding-top: 10px;
  
        &__submit {
          background: $submitColor;
          border: 1px solid rgba(0, 0, 0, 0.1);
          border-radius: .25rem;
          color: white;
          font-size: 1.2rem;
          padding: 6px 12px;
          cursor: pointer;
          
          &:disabled {
            opacity: .6;
            cursor: not-allowed;
          }
        }
      }
    }
  }

  @media only screen and (max-width: 600px) {
    :local .modal {
      &__form {
        &__error {
          &__message {
            font-size: .4rem;
          }
        }
      }
    }
}