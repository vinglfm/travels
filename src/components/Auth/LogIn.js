import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {logIn} from '../../actions';
import {connect} from 'react-redux';
import styles from './AuthModal.css';

class LogIn extends Component {
    constructor(props) {
        super(props);

        this.state = {
            email:'',
            password:''
        };

        this.emailValidationSchema = {
            email: { type: 'email' }
        };

        this.passwordValidationSchema = {
            password: { type: 'string', min: 5, max: 12 }
        }

        this.logIn = this.logIn.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    logIn(event) {
        this.props.logIn(this.state);
        this.props.onClose();
        event.preventDefault();
    }

    handleChange(elem) {
        this.setState({[elem.target.name]: elem.target.value});
    }

    render() {
        return (
            <div>
                <div className={styles.modal__header}>
                    <a onClick={this.props.onBack} className={styles.modal__back}><i className='fas fa-arrow-left'></i></a>
                    <span>Log In</span>
                </div>
                <hr className={styles.modal__divider}/>
                <form className={styles.modal__form} name='LogInForm' onSubmit={this.logIn} noValidate>
                    <label className={styles.modal__form__label}>Email<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={styles.modal__form__input} name='email' type='email' onChange={this.handleChange}/>
                    <label className={styles.modal__form__label}>Password<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={styles.modal__form__input} name='password' type='password' placeholder='Password' onChange={this.handleChange}/>
                    <div className={styles.modal__form__group}>
                        <input className={styles.modal__form__group__submit} type='submit' value='Log In'/>
                    </div>
                </form>
            </div>
        );
    }
}

LogIn.propTypes = {
    onBack: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired
};

const mapDispatchToProps = {
    logIn
};

export default connect(null, mapDispatchToProps)(LogIn);