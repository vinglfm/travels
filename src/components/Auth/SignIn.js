import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Validaton from '../../_common/Validation';
import actions from '../../_actions';
import { connect } from 'react-redux';
import styles from './AuthModal.css';

export class SignIn extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };

    this.validation = new Validaton();
    this.signIn = this.signIn.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  signIn(event) {
    this.props.signIn(this.state);
    this.props.onClose();
    event.preventDefault();
  }

  handleChange(elem) {
    this.setState({ [elem.target.name]: elem.target.value });
  }

  render() {
    const emailError = this.validation.validateEmail(this.state);
    const passwordError = this.validation.validatePassword(this.state);
    const emailErrorMessage = emailError[0] ? emailError[0].message : '';
    const passwordErrorMessage = passwordError[0] ? passwordError[0].message : '';
    return (
      <div>
        <div className={styles.modal__header}>
          <a onClick={this.props.onBack} className={styles.modal__back}><i className='fas fa-arrow-left'></i></a>
          <span>Sign In</span>
        </div>
        <hr className={styles.modal__divider} />
        <form className={styles.modal__form} name='SignInForm' onSubmit={this.signIn} noValidate>
          <label className={styles.modal__form__label}>Email<span className={styles.modal__form__label__required}>*</span></label>
          <input className={emailError === true || !this.state.email ? styles.modal__form__input : styles.modal__form__input__error} name='email' type='email' placeholder='Email' onChange={this.handleChange} />
          <div className={styles.modal__form__error}>
            <span className={`${styles.modal__form__error__message} ${!this.state.email ? styles.modal__form__error__message__hidden : ''}`}>{emailErrorMessage}</span>
          </div>
          <label className={styles.modal__form__label}>Password<span className={styles.modal__form__label__required}>*</span></label>
          <input className={passwordError === true || !this.state.password ? styles.modal__form__input : styles.modal__form__input__error} name='password' type='password' placeholder='Password' onChange={this.handleChange} />
          <div className={styles.modal__form__error}>
            <span className={`${styles.modal__form__error__message} ${!this.state.password ? styles.modal__form__error__message__hidden : ''}`}>{passwordErrorMessage}</span>
          </div>
          <div className={styles.modal__form__group}>
            <input className={styles.modal__form__group__submit} type='submit' disabled={emailError !== true || passwordError !== true} value='Sign In' />
          </div>
        </form>
      </div>
    );
  }
}

SignIn.propTypes = {
  onBack: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired
};

const mapDispatchToProps = {
  signIn: actions.signIn
};

export default connect(null, mapDispatchToProps)(SignIn);