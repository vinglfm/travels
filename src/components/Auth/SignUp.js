import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Validator from 'fastest-validator';
import styles from './AuthModal.css';

const validator = new Validator();

export default class SignUp extends Component {
    constructor(props) {
        super(props);

        this.state = {
            email:'',
            password:'',
            firstName: '',
            lastName: ''
        };

        this.emailValidationSchema = {
            email: { type: 'email' }
        };

        this.passwordValidationSchema = {
            password: { type: 'string', min: 5, max: 12 }
        }

        this.signUp = this.signUp.bind(this);
        this.validateEmail = this.validateEmail.bind(this);
        this.validatePassword = this.validatePassword.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    signUp() {
        console.log(this.state);
        this.props.onSignUp(this.state);
    }

    handleChange(elem) {
        this.setState({[elem.target.name]: elem.target.value});
        console.log(elem.target.name, '=', elem.target.value);
    }

    validateEmail() {
        return validator.validate(this.state, this.emailValidationSchema);
    }

    validatePassword() {
        return validator.validate(this.state, this.passwordValidationSchema);
    }

    render() {
        const emailValidationError = this.validateEmail();
        const passwordValidationError = this.validatePassword();
        const emailErrorMessage = emailValidationError[0] ? emailValidationError[0].message: '';
        const passwordErrorMessage = passwordValidationError[0] ? passwordValidationError[0].message: '';
        return (
            <div>                           
                <div className={styles.modal__header}>                    
                    <a onClick={this.props.baseComponent} className={styles.modal__back}><i className='fas fa-arrow-left'></i></a>
                    <span>Sign Up</span>
                </div>
                <hr className={styles.modal__divider}/>
                <form className={styles.modal__form} name='signUpForm' onSubmit={this.signUp} role='form' noValidate>
                    <label className={styles.modal__form__label}>Email<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={emailValidationError === true || !this.state.email ? styles.modal__form__input : styles.modal__form__input__error} name='email' type='email' placeholder='Email' onChange={this.handleChange}/>
                    <div className={styles.modal__form__error}>
                        <span className={`${styles.modal__form__error__message} ${!this.state.email ? styles.modal__form__error__message__hidden : ''}`}>{emailErrorMessage}</span>
                    </div>
                    <label className={styles.modal__form__label}>Password<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={passwordValidationError === true || !this.state.password ? styles.modal__form__input : styles.modal__form__input__error} name='password' type='password' placeholder='Password'onChange={this.handleChange}/>
                    <div className={styles.modal__form__error}>
                        <span className={`${styles.modal__form__error__message} ${!this.state.password ? styles.modal__form__error__message__hidden : ''}`}>{passwordErrorMessage}</span>
                    </div>
                    <label className={styles.modal__form__label}>Full Name</label>
                    <input className={styles.modal__form__input__small} name='firstName' type='text' placeholder='First' onChange={this.handleChange}/>&nbsp;
                    <input className={styles.modal__form__input__small} name='lastName' type='text' placeholder='Last' onChange={this.handleChange}/>
                    <div className={styles.modal__form__group}>
                    <input className={styles.modal__form__group__submit} disabled={emailValidationError !== true || passwordValidationError !== true} type='submit' value='Create Account'/>
                    </div>
                </form>
            </div>
        );
    }
}

SignUp.propTypes = {
    baseComponent: PropTypes.func.isRequired
};