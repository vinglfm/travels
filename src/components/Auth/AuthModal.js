import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import SignIn from './SignIn.js';
import styles from './AuthModal.css';


export default class AuthModal extends Component {
    render() {
        return (
            <Dialog
            open={this.props.open}
            onEnter={this.baseComponent}
            onClose={this.props.handleClose}>
            <DialogTitle>
                <div className={styles.dialog__title}>Continue with</div>
            </DialogTitle>
            <DialogContent>
                <SignIn onSignIn={this.props.handleClose}/>
            </DialogContent>
          </Dialog>
        );
    }
}

AuthModal.propTypes = {
    open: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired
};