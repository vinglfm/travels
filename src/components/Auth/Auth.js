import React, {Component} from 'react';
import FacebookLogin from 'react-facebook-login';
import SignUp from './SignUp.js';
import SignIn from './SignIn.js';
import PropTypes from 'prop-types';
import styles from './AuthModal.css';
import {connect} from 'react-redux';
import config from '../../config';
import actions from '../../_actions';

export class Auth extends Component {
    constructor(props) {
        super(props);
        
        this.state = {
            component: 'Base'
        };

        this.facebookResponse = this.facebookResponse.bind(this);
        this.getComponent = this.getComponent.bind(this);
        this.renderComponent = this.renderComponent.bind(this);
        this.baseComponent = this.baseComponent.bind(this);
    }

    facebookResponse (resp) {
        this.props.fbSignIn(resp.accessToken, () => {
            this.props.onSignIn();
        });
    }
  
    renderComponent(component) {
        this.setState({
            component
        });
    }

    baseComponent() {
        return (
            <div className={styles.auth}>
                <FacebookLogin appId={config.appId}
                autoLoad={false} fields='name,email'
                cssClass={styles.fb__button} callback={this.facebookResponse}
                textButton='Facebook' />
                <div className={styles.lines}>
                    <span>or</span>
                </div>
                <div className={styles.sign__in}><a onClick={() => this.renderComponent('SignIn')}>Sign In</a></div>
                <div className={styles.sign__up}>First time travel? <a onClick={() => this.renderComponent('SignUp')}>Sign Up</a></div>
            </div>);
    }

    getComponent() {
        switch(this.state.component) {
            case 'SignIn':
                return <SignIn onBack={() => this.renderComponent('Base')} onClose={() => this.props.onSignIn()}/>;
            case 'SignUp':
                return <SignUp onBack={() => this.renderComponent('Base')} onClose={() => this.props.onSignIn()}/>;
            default:
                return this.baseComponent();
        }
    }

    render() {
        return (<div>
            {this.getComponent()}
            </div>);
    }
}

Auth.propTypes = {
    onSignIn: PropTypes.func.isRequired 
};

const mapDispatchToProps = {
    fbSignIn: actions.fbSignIn
};

export default connect(null, mapDispatchToProps)(Auth);