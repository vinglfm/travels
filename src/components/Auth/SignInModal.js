import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import LogIn from './LogIn.js';
import SignIn from './SignIn.js';
import SignUp from './SignUp.js';
import styles from './SignInModal.css';


export default class SignInModal extends Component {
    constructor(props) {
        super(props);
        this.state = {
            component: 'SignIn'
        };

        this.getComponent = this.getComponent.bind(this);
        this.logInComponent = this.logInComponent.bind(this);
        this.signUpComponent = this.signUpComponent.bind(this);
        this.baseComponent = this.baseComponent.bind(this);
    }

    logInComponent(componentName) {
        this.setState({
            component: 'LogIn'
        });
    }

    signUpComponent(componentName) {
        this.setState({
            component: 'SignUp'
        });
    }

    baseComponent() {
        this.setState({
            component: 'SignIn'
        });
    }
  
    getComponent() {
        switch(this.state.component) {
            case 'LogIn':
                return <LogIn baseComponent={this.baseComponent}/>;
            case 'SignUp':
                return <SignUp baseComponent={this.baseComponent}/>;
            default:
                return <SignIn logInComponent={this.logInComponent} signUpComponent={this.signUpComponent} onSignIn={this.props.handleClose}/>;
        }
    }

    render() {
        return (
            <Dialog
            open={this.props.open}
            onEnter={this.baseComponent}
            onClose={this.props.handleClose}>
            <DialogTitle>
                <div className={styles.dialog__title}>Continue with</div>
            </DialogTitle>
            <DialogContent>
                {this.getComponent()}
            </DialogContent>
          </Dialog>
        );
    }
}

SignInModal.propTypes = {
    open: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired
};