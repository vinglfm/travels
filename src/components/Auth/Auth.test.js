import enzyme, {shallow} from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new ReactSixteenAdapter() });

import React from 'react';
import {Auth} from './Auth';

it('renders Auth', () => {
  const snapshot = shallow(<Auth onSignIn={()=>{}} />);
  expect(snapshot).toMatchSnapshot();
});

it('renders SignIn component when state is chagned to LogIn', () => {
  const snapshot = shallow(<Auth onSignIn={()=>{}} />);
  snapshot.instance().renderComponent('SignIn');
  snapshot.update();
  expect(snapshot).toMatchSnapshot();
});

it('renders SignUp component when state is changed to SignUp', () => {
  const snapshot = shallow(<Auth onSignIn={()=>{}} />);
  snapshot.instance().renderComponent('SignUp');
  snapshot.update();
  expect(snapshot).toMatchSnapshot();
});
