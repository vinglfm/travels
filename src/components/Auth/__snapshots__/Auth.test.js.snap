// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Auth 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Auth
    onSignIn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
>
        <t
                appId="1817436878556969"
                autoLoad={false}
                callback={[Function]}
                cssClass="kep-login-facebook"
                fields="name,email"
                size="metro"
                tag="button"
                textButton="Facebook"
                typeButton="button"
        />
        <div
                className={undefined}
        >
                <span>
                        or
                </span>
        </div>
        <div
                className={undefined}
        >
                <a
                        onClick={[Function]}
                >
                        Sign In
                </a>
        </div>
        <div
                className={undefined}
        >
                First time travel? 
                <a
                        onClick={[Function]}
                >
                        Sign Up
                </a>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <t
            appId="1817436878556969"
            autoLoad={false}
            callback={[Function]}
            cssClass="kep-login-facebook"
            fields="name,email"
            size="metro"
            tag="button"
            textButton="Facebook"
            typeButton="button"
/>,
          <div
            className={undefined}
>
            <span>
                        or
            </span>
</div>,
          <div
            className={undefined}
>
            <a
                        onClick={[Function]}
            >
                        Sign In
            </a>
</div>,
          <div
            className={undefined}
>
            First time travel? 
            <a
                        onClick={[Function]}
            >
                        Sign Up
            </a>
</div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appId": "1817436878556969",
            "autoLoad": false,
            "callback": [Function],
            "cssClass": "kep-login-facebook",
            "fields": "name,email",
            "size": "metro",
            "tag": "button",
            "textButton": "Facebook",
            "typeButton": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              or
</span>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "or",
            },
            "ref": null,
            "rendered": "or",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
>
              Sign In
</a>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "First time travel? ",
              <a
                onClick={[Function]}
>
                Sign Up
</a>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            "First time travel? ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
>
          <t
                    appId="1817436878556969"
                    autoLoad={false}
                    callback={[Function]}
                    cssClass="kep-login-facebook"
                    fields="name,email"
                    size="metro"
                    tag="button"
                    textButton="Facebook"
                    typeButton="button"
          />
          <div
                    className={undefined}
          >
                    <span>
                              or
                    </span>
          </div>
          <div
                    className={undefined}
          >
                    <a
                              onClick={[Function]}
                    >
                              Sign In
                    </a>
          </div>
          <div
                    className={undefined}
          >
                    First time travel? 
                    <a
                              onClick={[Function]}
                    >
                              Sign Up
                    </a>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <t
              appId="1817436878556969"
              autoLoad={false}
              callback={[Function]}
              cssClass="kep-login-facebook"
              fields="name,email"
              size="metro"
              tag="button"
              textButton="Facebook"
              typeButton="button"
/>,
            <div
              className={undefined}
>
              <span>
                            or
              </span>
</div>,
            <div
              className={undefined}
>
              <a
                            onClick={[Function]}
              >
                            Sign In
              </a>
</div>,
            <div
              className={undefined}
>
              First time travel? 
              <a
                            onClick={[Function]}
              >
                            Sign Up
              </a>
</div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "appId": "1817436878556969",
              "autoLoad": false,
              "callback": [Function],
              "cssClass": "kep-login-facebook",
              "fields": "name,email",
              "size": "metro",
              "tag": "button",
              "textButton": "Facebook",
              "typeButton": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                or
</span>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "or",
              },
              "ref": null,
              "rendered": "or",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
>
                Sign In
</a>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "a",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "First time travel? ",
                <a
                  onClick={[Function]}
>
                  Sign Up
</a>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              "First time travel? ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Up",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders SignIn component when state is chagned to LogIn 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Auth
    onSignIn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(SignIn)
        onBack={[Function]}
        onClose={[Function]}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onBack": [Function],
        "onClose": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(SignIn)
          onBack={[Function]}
          onClose={[Function]}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onBack": [Function],
          "onClose": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders SignUp component when state is changed to SignUp 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Auth
    onSignIn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(SignUp)
        onBack={[Function]}
        onClose={[Function]}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onBack": [Function],
        "onClose": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(SignUp)
          onBack={[Function]}
          onClose={[Function]}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onBack": [Function],
          "onClose": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
